SCClassLibrary/Common/Streams/PmonoStreams.sc

```diff
@@ -69,7 +69,7 @@ PmonoStream : Stream {
 					~type = \monoNote;
 					~instrument = pattern.synthName;
 					cleanup.addFunction(event, currentCleanupFunc = { | flag |
-						if (flag) { (id: id, server: server, type: \off,
+						if (flag && id.notNil) { (id: id, server: server, type: \off,
```

Comment:
One minor suggestion: if it's written `flag and: { id.notNil }` then there are two benefits: 1/ save one method dispatch (`&&` redirects to `and` -- unfortunately `&&` is not optimized as it is in C) and 2/ the second condition is short-circuited when `flag` is false. I'm pretty sure the cleanup is almost always called with `flag` being true, so short-circuiting isn't a big deal, but saving one method lookup is better (because we don't know how many events per second users may be generating, and how often they will rearticulate).

Reply:
Flag is false only when you hit cmd+. (CmdPeriod executed), as far as I know. So in the vast majority of cases/calls there will be no short-circuit, but your point is taken.